core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -93,6 +93,16 @@ class SparkHadoopUtil extends Logging {
         hadoopConf.set("fs.s3.awsSecretAccessKey", accessKey)
         hadoopConf.set("fs.s3n.awsSecretAccessKey", accessKey)
         hadoopConf.set("fs.s3a.secret.key", accessKey)
+
+        if (System.getenv("AWS_SESSION_TOKEN") != null) {
+          val sessionToken = System.getenv("AWS_SESSION_TOKEN")
+          hadoopConf.set("fs.s3a.session.token", sessionToken)
+          logDebug(s"Found 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY' and " +
```

Comment:
Are these debug statements that useful? they don't need to be interpolated strings, FWIW.